scavenge/tutorial/05-types.md

```diff
@@ -0,0 +1,48 @@
+---
+order: 5
+---
+
+# Types
+
+Now that you've defined messages that trigger state transitions, it's time to implement types and methods that operate on the state.
+
+A keeper is an abstraction that let's your blockchain app interact with the state. Functions like create, update and delete are defined as keeper methods. In the Scavenge blockchain a `scavenge` and `commit` types need to be defined, along with create and update methods.
+
+Starport has several commands that scaffold the code for CRUD functionality for a list-like data structure, a map (key-value pairs) and a single element in the state. In this example, both `scavenge` and `commit` will be stored in a map-like data structure.
+
+## Scavenge
+
+Use `starport scaffold map` command to scaffold the `scavenge` type and the code for creating, reading, updating, and deleting (CRUD) scavenges. The first argument is the name of the type being created (`scavenge`), the rest is list of fields. By default, generic CRUD messages are scaffolded, but since you've already created messages specifically for this blockchain, skip messages with a `--no-message` flag.
+
+```
+starport scaffold map scavenge solutionHash solution description reward scavenger --no-message
```

Comment:
I get this error here 

```bash
> starport scaffold map scavenge solutionHash solution description reward scavenger --no-message

component type with name scavenge is already created (type MsgCreateScavenge exists)
```

Reply:
True, this is a bug.

Reply:
https://github.com/tendermint/starport/issues/1379

Reply:
For this not to block us, I propose we put the Keeper section before Messages, so that `map scavenge ... --no-message` will happen before message scaffolding.

Reply:
Alternatively, we can rename `create-scavenge` with `submit-scavenge`. (until the bug is fixed, of course)

Reply:
Any downside to the renaming?

Reply:
No, it's just going to change the command in the Play section. Let me know if you prefer renaming the message (which is easier), and I'll take care of it.

Reply:
Yes, let's rename.

Reply:
Should be fixed.

Reply:
We will actually a better way to detect already scaffolded component and take `--no-message` into account

Although, in this case, isn't more intuitive to first scaffold the type before scaffolding the message to interact with this type?

Reply:
It depends. Messages represent actions that a user can do (through CLI commands), so for some, it's easier to start from these actions, then scaffold CRUD methods for a type, and then wire everything together.